spec/utils/version_control/git_annex_spec.rb

```diff
@@ -38,4 +62,231 @@
+    let(:cloned_repo_path) { git_annex.clone }
+    let(:git) { ExtendedGit.open(cloned_repo_path) }
+    let(:first_new_file) { 'first_new_file.txt' }
+    let(:second_new_file) { 'second_new_file.txt' }
+
+    before do
+      [first_new_file, second_new_file].each do |new_file|
+        new_file_path = File.join(cloned_repo_path, new_file )
+        # Each file needs to have different content.
+        File.open(new_file_path, 'w') { |f| f.write("New file -- #{new_file}") }
+        git.add(new_file_path)
+      end
+
+      git_annex.commit('New files.', cloned_repo_path)
+    end
+
+    after { git_annex.remove_working_directory(cloned_repo_path) }
+
+    context 'when pushing all new content' do
+      before { git_annex.push({}, cloned_repo_path) }
+
+      it 'content is stored in special remote' do
+        expect(git.annex.whereis[first_new_file].locations.map(&:description)).to include '[local]'
+        expect(git.annex.whereis[second_new_file].locations.map(&:description)).to include '[local]'
+      end
+    end
+
+    context 'when only pushing specified content' do
+      before do
+        git_annex.push({ content: second_new_file }, cloned_repo_path)
+      end
+
+      it 'expected content is stored in special remote' do
+        expect(git.annex.whereis[second_new_file].locations.map(&:description)).to include '[local]'
+      end
+
+      it 'expected content is NOT stored in special remote' do
+        expect(git.annex.whereis[first_new_file].locations.map(&:description)).not_to include '[local]'
+      end
+    end
+  end
+
+  describe '#drop' do
+    let(:cloned_repo_path) { git_annex.clone }
+    let(:git) { ExtendedGit.open(cloned_repo_path) }
+    let(:new_file) { 'new_file.txt' }
+
+    # Adding an additional file.
+    before do
+      new_file_path = File.join(cloned_repo_path, new_file)
+      FileUtils.touch(new_file_path)
+      git.add(new_file_path)
+      git_annex.commit('Adding new file.', cloned_repo_path)
+      git_annex.push({}, cloned_repo_path)
+    end
+
+    after { git_annex.remove_working_directory(cloned_repo_path) }
+
+    # Checking that files are present in working directory.
+    it 'contains files in current working directory' do
+      expect(git.annex.whereis.all?(&:here?)).to be true
+    end
+
+    context 'when dropping everything' do
+      before { git_annex.drop({}, cloned_repo_path) }
+
+      it 'drops all files' do
+        expect(git.annex.whereis.any?(&:here?)).to be false
+      end
+    end
+
+    context 'when dropping specific content' do
+      let(:filename) { 'README.md' }
+
+      before { git_annex.drop({ content: filename }, cloned_repo_path) }
+
+      it 'only drops README.md' do
+        expect(git.annex.whereis[filename].here?).to be false # README.md dropped.
+        expect(git.annex.whereis[new_file].here?).to be true # Other files NOT dropped.
+      end
+    end
+
+    context 'when imaging user configured' do
+      it 'changes permissions'
```

Comment:
does this pass or fail? maybe you need `xit` here

Reply:
It's just listed as pending in the output. I could add a more descriptive pending message.